tests/modeltests/modeltest.cpp

```diff
@@ -342,7 +342,8 @@ void ModelTest::checkChildren(const QModelIndex &parent, int currentDepth)
             Q_ASSERT(index.column() == c);
             // While you can technically return a QVariant usually this is a sign
             // of an bug in data()  Disable if this really is ok in your model.
-            Q_ASSERT(model->data(index, Qt::DisplayRole).isValid() == true);
+            Q_ASSERT(model->data(index, Qt::DisplayRole).isValid() == true ||
```

Comment:
no that sounds wrong, and actually we should remove that copy of modeltest (I'll do that now) - we should use the upstream QAbstractItemModelTester now

if you return invalid display role but a valid tooltip role, that sounds wrong - fix the model